catroid/src/androidTestPhiro/java/org/catrobat/catroid/uiespresso/ui/activity/PhiroSettingsActivityTest.java

```diff
@@ -0,0 +1,172 @@
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.intent.Intents.intended;
+import static android.support.test.espresso.intent.Intents.intending;
+import static android.support.test.espresso.intent.matcher.IntentMatchers.hasAction;
+import static android.support.test.espresso.intent.matcher.IntentMatchers.hasData;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+
+import static junit.framework.Assert.assertFalse;
+import static junit.framework.Assert.assertTrue;
+
+import static org.catrobat.catroid.ui.SettingsActivity.SETTINGS_CAST_GLOBALLY_ENABLED;
+import static org.catrobat.catroid.ui.SettingsActivity.SETTINGS_CRASH_REPORTS;
+import static org.catrobat.catroid.ui.SettingsActivity.SETTINGS_MINDSTORMS_EV3_BRICKS_ENABLED;
+import static org.catrobat.catroid.ui.SettingsActivity.SETTINGS_MINDSTORMS_EV3_SHOW_SENSOR_INFO_BOX_DISABLED;
+import static org.catrobat.catroid.ui.SettingsActivity.SETTINGS_MINDSTORMS_NXT_BRICKS_ENABLED;
+import static org.catrobat.catroid.ui.SettingsActivity.SETTINGS_MINDSTORMS_NXT_SHOW_SENSOR_INFO_BOX_DISABLED;
+import static org.catrobat.catroid.ui.SettingsActivity.SETTINGS_SHOW_ARDUINO_BRICKS;
+import static org.catrobat.catroid.ui.SettingsActivity.SETTINGS_SHOW_HINTS;
+import static org.catrobat.catroid.ui.SettingsActivity.SETTINGS_SHOW_NFC_BRICKS;
+import static org.catrobat.catroid.ui.SettingsActivity.SETTINGS_SHOW_PARROT_AR_DRONE_BRICKS;
+import static org.catrobat.catroid.ui.SettingsActivity.SETTINGS_SHOW_PHIRO_BRICKS;
+import static org.catrobat.catroid.ui.SettingsActivity.SETTINGS_SHOW_RASPI_BRICKS;
+import static org.hamcrest.Matchers.allOf;
+
+@RunWith(AndroidJUnit4.class)
+public class PhiroSettingsActivityTest {
+
+	@Rule
+	public BaseActivityInstrumentationRule<PhiroSettingsActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(PhiroSettingsActivity.class, true, false);
+
+	private List<String> allSettings = new ArrayList<>(Arrays.asList(
+			SETTINGS_SHOW_ARDUINO_BRICKS, SETTINGS_SHOW_NFC_BRICKS, SETTINGS_SHOW_HINTS, SETTINGS_CRASH_REPORTS,
+			SETTINGS_MINDSTORMS_NXT_BRICKS_ENABLED, SETTINGS_MINDSTORMS_NXT_SHOW_SENSOR_INFO_BOX_DISABLED,
+			SETTINGS_MINDSTORMS_EV3_BRICKS_ENABLED, SETTINGS_MINDSTORMS_EV3_SHOW_SENSOR_INFO_BOX_DISABLED,
+			SETTINGS_SHOW_PARROT_AR_DRONE_BRICKS, SETTINGS_SHOW_RASPI_BRICKS, SETTINGS_CAST_GLOBALLY_ENABLED));
+
+	private Map<String, Boolean> initialSettings = new HashMap<>();
+
+	@Before
+	public void setUp() throws Exception {
+		SharedPreferences sharedPreferences = PreferenceManager
+				.getDefaultSharedPreferences(InstrumentationRegistry.getTargetContext());
+
+		for (String setting : allSettings) {
+			initialSettings.put(setting, sharedPreferences.getBoolean(setting, false));
+		}
+		initialSettings.put(SETTINGS_SHOW_PHIRO_BRICKS, sharedPreferences.getBoolean(SETTINGS_SHOW_PHIRO_BRICKS, false));
```

Comment:
if the test is about checking out the default / initial state, why touch shared preferences in the first place? 
also this should be verified via UI, since you have to give a fallback when asking sharedPreferences, and if you do sharedPreferences.getBoolean(SETTINGS_SHOW_PHIRO_BRICKS, false) in your test, doesnt mean that someone might have used sharedPreferences.getBoolean(SETTINGS_SHOW_PHIRO_BRICKS, true) in production code, (would make a difference if run for the first time and key isnt there yet)